thrust/system/cuda/detail/unique.h

```diff
@@ -794,6 +804,36 @@ unique(execution_policy<Derived> &policy,
   return cuda_cub::unique(policy, first, last, equal_to<input_type>());
 }
 
+
+template <typename BinaryPred>
+struct zip_adj_not_predicate {
+  template <typename TupleType>
+  bool __host__ __device__ operator()(TupleType&& tuple) {
+      return !binary_pred(thrust::get<0>(tuple), thrust::get<1>(tuple));
+  }
+  
+  BinaryPred binary_pred;
+};
+
+
+__thrust_exec_check_disable__
+template <class Derived,
+          class InputIt,
+          class BinaryPred>
+typename thrust::iterator_traits<InputIt>::difference_type __host__ __device__
+unique_count(execution_policy<Derived> &policy,
+       InputIt                    first,
+       InputIt                    last,
+       BinaryPred                 binary_pred)
+{
+  if (first == last) {
+    return 0;
+  }
+  auto size = last - first;
+  auto it = thrust::make_zip_iterator(thrust::make_tuple(first, first + 1));
+  return 1 + thrust::count_if(policy, it, it + (size - 1), zip_adj_not_predicate<BinaryPred>{binary_pred});
```

Comment:
What I haven't looked in deeper here: What if policy is a device policy, but first and last host pointers? I'm guessing that means that Thrust would create a temporary copy of the *tuples* on the device? That's probably not desirable

Reply:
The answer is: don't do that. If you pass a device policy and host pointers, it blows up. Not supported.

Reply:
Thanks for the clarification, I thought I saw that working somewhere in the past